packages/acter_notifify/lib/processing/utils.dart

```diff
@@ -1,31 +1,34 @@
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
 
-String parentPart(ActivityObject parent) {
-  final emoji = parent.emoji();
-  final title = switch (parent.typeStr()) {
-    'news' => "boost",
-    _ => parent.title(),
-  };
-  return "$emoji $title";
-}
-
-String getUserCentricTitlePart(
-  NotificationItem item,
-  String? emoji,
-  String suffix,
-) {
-  final sender = item.sender();
-  final username = sender.displayName() ?? sender.userId();
+extension ActerNotififyActivityObjectExtension on ActivityObject {
+  String? parentPart() {
+    final emoji = this.emoji();
+    final title = switch (typeStr()) {
+      'news' => 'boost',
+      'story' => 'story',
+      _ => this.title(),
+    };
+    return title != null ? "$emoji $title" : null;
+  }
 
-  String titlePart = '$username $suffix';
-  if (emoji != null) titlePart = '$emoji $titlePart';
-  return titlePart;
+  String? getObjectCentricTitlePart(
+    String suffix,
+  ) {
+    final parentInfo = parentPart();
+    return parentInfo != null ? '$parentInfo $suffix' : null;
+  }
```

Comment:
and with `parentPart` nullable we can have both of these just also return null if we have no title and make the actual code go back to the fallback cases.